lib/sycamore/sycamore/transforms/detr_partitioner.py

```diff
@@ -666,8 +664,52 @@ def infer(self, images: List[Image.Image], threshold: float) -> List[List[Elemen
                 )
                 elements.append(element)
             batched_results.append(elements)
+            if self.cache:
+                key = Cache.get_hash_key(image.tobytes())
+                self.cache.set(key, result)
+
         return batched_results
 
+    def _get_cached_inference(self, images: List[Image.Image], threshold: float) -> list:
+        results = []
+        uncached_images = []
+        uncached_indices = []
+
+        # First, check the cache for each image
+        for index, image in enumerate(images):
+            key = Cache.get_hash_key(image.tobytes())
+            cached_image = self.cache.get(key)
+            if cached_image:
+                print("Cache hit. Returning the result from cache")
+                results.append(cached_image)
+            else:
+                uncached_images.append(image)
+                uncached_indices.append(index)
```

Comment:
Rather than these parallel data structures, I'd use a list of tuples: (index, image).  That said, I don't think you even need that.  Just run through the results array and anywhere you have a None, you need to do the actual work.  You already have the images array as a parameter, and the indices will be the same in the two arrays, right?

Reply:
I want to re-use the logic in _get_uncached_inference! Also, if the batch size changes to be always 1 (As per Eric's recommendation) this logic will get simplified.

Reply:
I see it.